src/Utilities/Tuple.hpp

```diff
@@ -204,3 +204,19 @@ auto tuple_tail(const std::tuple<AllElements...>& tuple) noexcept {
   static constexpr size_t full_size = sizeof...(AllElements);
   return tuple_slice<full_size - Size, full_size>(tuple);
 }
+
+namespace tuple_impl_detail {
+
```

Comment:
do we need the blank lines at the start and end of the namespaces?

Reply:
I find that more readable 🤷‍♂️ helps to see where namespaces start and end, and where classes and functions start, imo